tools/odbc/statement_functions.cpp

```diff
@@ -97,11 +97,14 @@ static void LogInvalidCast(const LogicalType &from_type, const LogicalType &to_t
 
 template <class T>
 static SQLRETURN GetInternalValue(duckdb::OdbcHandleStmt *stmt, const duckdb::Value &val, const LogicalType &type,
-                                  SQLLEN buffer_length, SQLPOINTER target_value_ptr) {
+                                  SQLLEN buffer_length, SQLPOINTER target_value_ptr, SQLLEN *str_len_or_ind_ptr) {
 	D_ASSERT(((size_t)buffer_length) >= sizeof(T));
 	try {
 		auto casted_value = val.CastAs(type).GetValue<T>();
 		Store<T>(casted_value, (duckdb::data_ptr_t)target_value_ptr);
+		if (str_len_or_ind_ptr) {
```

Comment:
Should this be here still?

Reply:
I think so. We need to check if a pointer to the length of the data was provided, then fill it with the number of bytes written: 

"_For all other data types, this is the length of the data after conversion; that is, it is the size of the type to which the data was converted._" 

 https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetdata-function?view=sql-server-ver15#:~:text=For%20all%20other%20data%20types%2C%20this%20is%20the%20length%20of%20the%20data%20after%20conversion%3B%20that%20is%2C%20it%20is%20the%20size%20of%20the%20type%20to%20which%20the%20data%20was%20converted.

Reply:
Apologies, this wasn't entirely clear. I meant the fact that the line after this was commented out:

```cpp
if (str_len_or_ind_ptr) {
  // *str_len_or_ind_ptr = sizeof(T);
}
```

Reply:
I see, I forgot to remove the comment symbols (//). I changed in my branch, but I don't commit yet.